src/Models/Dspace.php

```
@@ -0,0 +1,238 @@
+            $action = $queryParams->getQuery()->getString('dspace_action');
+        }
+        return match ($action) {
+            'auth' => $this->getDspaceToken(),
+            'collections' => $this->listCollections(),
+            'types' => $this->listTypes(),
+            default => throw new ImproperActionException('Unknown DSpace GET action.'),
+        };
+    }
+
+    #[Override]
+    public function postAction(Action $action, array $reqBody): int
```

PHPMD (2.15.0)

Avoid unused parameters such as '$action'. (undefined)

(UnusedFormalParameter)